### PR TITLE
🛡️ Elenchus: Strengthen WalRingBuffer safety tests

### PR DESCRIPTION
**Pull request by madmax983**
Strengthened the test suite for `WalRingBuffer` by adding targeted safety tests for drop behavior, sequence gaps, and wraparound boundaries. This ensures that the WAL ring buffer handles edge cases correctly and prevents potential deadlocks or data corruption.

---
*PR created automatically by Jules for task [5750567160305692419](https://jules.google.com/task/5750567160305692419) started by @madmax983*

### CODE REVIEW

File: src/storage/wal/ring_buffer.rs [hunk 1]
```diff
@@ -1457,4 +1457,133 @@ mod sentry_tests {
...
+        let mut buf = WalRingBuffer::new(4);
+
+        // Initialize state: write_pos=2, read_pos=0.
+        // set_state_for_wraparound_test initializes slots to "Available for Write" state.
+        // For Slot 0 (pos 0) this means seq=0.
+        // For Slot 1 (pos 1) this means seq=1.
+        //
+        // Expected seq for drain(read_pos=0) is 1.
+        // Expected seq for drain(read_pos=1) is 2.
+        buf.set_state_for_wraparound_test(2, 0);
```
[the remark this answers is not in the document]
Comment (madmax983) [reply src/storage/wal/ring_buffer.rs:1518]: > **P2 Badge Build the gap test with a real unread-slot sequence**
> 
> `test_drain_stops_at_gap` initializes state with `set_state_for_wraparound_test(2, 0)`, but that helper sets slot sequences for positions starting at `write_pos`, so for capacity 4 the slot sequences become `[4, 5, 2, 3]` rather than the commented `[0, 1, ...]`. This means the first drain assertion is exercising an impossible mixed state (`current_seq > expected_seq`) instead of the intended gap case (`current_seq < expected_seq`), so a regression in how `drain` handles true missing writes could pass this test.
> 
> Useful? React with 👍 / 👎.

@jules

File: src/storage/wal/ring_buffer.rs [hunk 2]
```diff
@@ -1457,4 +1457,133 @@ mod sentry_tests {
...
+        // 🛡️ Sentry: Verify try_append logic exactly at the u64::MAX boundary.
+        // Specifically check the logic `distance_behind <= capacity`.
+
+        let capacity = 4;
+        let mut buf = WalRingBuffer::new(capacity);
+
+        // Set state to u64::MAX.
+        // We want write_pos = u64::MAX.
+        let boundary = u64::MAX;
+        buf.set_state_for_wraparound_test(boundary, boundary);
```
[the remark this answers is not in the document]
Comment (madmax983) [reply src/storage/wal/ring_buffer.rs:1577]: > **P2 Badge Exercise distance_behind branch in wraparound test**
> 
> `test_wraparound_boundary_check` claims to validate the `distance_behind <= capacity` logic, but with `set_state_for_wraparound_test(boundary, boundary)` the target slot's sequence equals `expected_seq`, so `try_append` takes the fast `current_seq == expected_seq` path and never evaluates the `distance_behind` branch. As written, this test cannot catch regressions in the boundary condition it documents.
> 
> Useful? React with 👍 / 👎.

@jules
